File: UIPeriodicTable/CPlusPlus/Resources/Components/code_uibargraph.cpp
// UIBargraph application source code
#include "PeriodicApp.h"

void PeriodicApp::setup_uibargraph( UIWidget *layout )
{
    PODVector<UIWidget*> dest;
    layout->SearchWidgetClass( "TBButton", dest );
    for (unsigned ii = 0; ii < dest.Size(); ii++)
        SubscribeToEvent(dest[ii], E_WIDGETEVENT, ATOMIC_HANDLER(PeriodicApp,HandleUibargraphEvent ));
}

void PeriodicApp::HandleUibargraphEvent(StringHash eventType, VariantMap& eventData)
{
    using namespace WidgetEvent;
    UIWidget* widget = static_cast<UIWidget*>(eventData[P_TARGET].GetPtr());
    if ( widget == NULL ) return;
    if (eventData[P_TYPE] == UI_EVENT_TYPE_CLICK)
    {
        if (widget->GetId() == "uibargraphcode" )
        {
            AppLog( "UIBargraph support : " + widget->GetId() + " was pressed " );
            ViewCode ( "Components/code_uibargraph.cpp", widget->GetParent() );
        }
        if (widget->GetId() ==  "uibargraphlayout" )
        {
            AppLog( "UIBargraph support : " + widget->GetId() + " was pressed ");
            ViewCode ( "Scenes/layout_uibargraph.ui.txt", widget->GetParent() );
        }
    }
}

File: UIPeriodicTable/CSharp/Resources/Components/code_uibargraph.cs
// UIBargraph application source code
using System;
using AtomicEngine;

public class code_uibargraph  : CSComponent {

    public void Setup( UIWidget layout )
    {
        var dest = new AtomicEngine.Vector<AtomicEngine.UIWidget>();
        layout.SearchWidgetClass( "TBButton", dest );
        for (var ii = 0; ii < dest.Size; ii++) {  // set bulk event handlers on all buttons -- boom!
            dest[ii].SubscribeToEvent<WidgetEvent> (dest [ii], HandleUibargraphEvent );
        }
    }

    private static void HandleUibargraphEvent( WidgetEvent ev )
    {
        UIWidget widget = (UIWidget)ev.Target;
        if ( widget.Equals(null)) return;
        if ( ev.Type == UI_EVENT_TYPE.UI_EVENT_TYPE_CLICK) {
            if (widget.GetId() == "uibargraphcode" ) {
                AtomicMain.AppLog( "UIBargraph support : " + widget.GetId() + " was pressed " );
                AtomicMain.ViewCode ( "Components/code_uibargraph.cs", widget.GetParent() );
            }
            if (widget.GetId() ==  "uibargraphlayout" ) {
                AtomicMain.AppLog( "UIBargraph support : " + widget.GetId() + " was pressed ");
                AtomicMain.ViewCode ( "Scenes/layout_uibargraph.ui.txt", widget.GetParent() );
            }
        }
    }
}

